src/UI/Actions.hs

```diff
@@ -0,0 +1,246 @@
+{-# LANGUAGE OverloadedStrings #-}
+module UI.Actions
+       (backToIndex, haltApp, focusSearch, displayMail, setUnread,
+        applySearchTerms, mailIndexUp, mailIndexDown, switchComposeEditor,
+        composeMail, replyMail, scrollUp, scrollDown, toggleHeaders, send,
+        reset, updateStateWithParsedMail, updateReadState, initialCompose)
+       where
+
+import Brick.Main (continue, halt, vScrollPage)
+import qualified Brick.Types as T
+import qualified Brick.Widgets.Edit as E
+import qualified Brick.Widgets.List as L
+import Network.Mail.Mime (Address(..), renderSendMail, simpleMail')
+import Data.Semigroup ((<>))
+import Data.Vector (Vector)
+import Data.Text (unlines)
+import Data.Text.Lazy.IO (readFile)
+import Prelude hiding (readFile, unlines)
+import Control.Lens (set, over, view, (&), _Just)
+import Control.Lens.Fold ((^?!))
+import Control.Monad ((>=>))
+import Control.Monad.Except (runExceptT)
+import Control.Monad.IO.Class (liftIO)
+import Data.Text.Zipper (currentLine, gotoEOL)
+import Data.Text (Text)
+import Storage.Notmuch (getMessages, addTag, removeTag, setNotmuchMailTags)
+import Storage.ParsedMail (parseMail, getTo, getFrom, getSubject)
+import Types
+import Error
+
+backToIndex :: Action a
+backToIndex =
+    Action
+    { _aDescription = "back to the index"
+    , _aAction = continue . set asAppMode BrowseMail
+    }
+
+haltApp :: Action a
+haltApp =
+    Action
+    { _aDescription = "quit the application"
+    , _aAction = halt
+    }
+
+composeMail :: Action (L.List Name NotmuchMail)
```

Comment:
This implies that we can only begin composing a mail from the mail index?
Fine for now - I'm just checking my mental model and suggesting that we might
want to generalise it **at a future point**.

Reply:
Yes currently that's the case. I ran into a situation today tho, that I think generalising it would make sense.